javalin/src/main/java/io/javalin/jetty/JettyResourceHandler.kt

```diff
@@ -43,7 +43,10 @@ class JettyResourceHandler(val pvt: PrivateConfig) : JavalinResourceHandler {
                 if (resource.isFile() || resource.isDirectoryWithWelcomeFile(handler, target)) {
                     handler.config.headers.forEach { httpResponse.setHeader(it.key, it.value) }
                     if (handler.config.precompress) {
-                        return JettyPrecompressingResourceHandler.handle(target, resource, httpRequest, httpResponse)
+                        return if (resource.isDirectoryWithWelcomeFile(handler, target))  // if it's a directory, we need to serve the welcome file
+                            JettyPrecompressingResourceHandler.handle(target, getWelcomeFile(handler, target), httpRequest, httpResponse)
+                        else
+                            JettyPrecompressingResourceHandler.handle(target, resource, httpRequest, httpResponse)
```

Comment:
This looks like something that should rather be handled inside the `JettyPrecompressingResourceHandler`, since both branches of the if send you there.

Reply:
I thought about doing so, but discarded it in favour of having all Welcome File logic/checks in a single class and exempt the precompressing class from this responsibility and just handle files. But if you want to I'll change it!

Reply:
Yes, please move it to the other handler, I think it's cleaner that way. Ideally we should move more logic to that handler, so that we don't call it from the "normal" handler, but that's for the future.

Reply:
I've just realized this would be a major refactor. `JettyPrecompressingResourceHandler` does not implement `ResourceHandler`, thus making it impossible to use the `getResource` method.

Reply:
Okay, then we keep it like this.